pallets/connectors/src/lib.rs

```diff
@@ -0,0 +1,507 @@
+			// is a MemberListAdmin and thus allowed to add other members.
+			ensure!(
+				T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					who.clone(),
+					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, Self::now()))
+				) || T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					who.clone(),
+					Role::PoolRole(PoolRole::MemberListAdmin)
+				),
+				BadOrigin
+			);
+
+			T::Permission::add(
+				PermissionScope::Pool(pool_id),
+				address.into_account_truncating(),
+				Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, valid_until)),
+			)?;
+
+			Self::do_send_message(
+				who,
+				Message::UpdateMember {
+					pool_id,
+					tranche_id,
+					valid_until,
+					address: address.address,
+				},
+				address.domain,
+			)?;
+
+			Ok(())
+		}
+
+		/// Transfer tranche tokens to a given address
+		#[pallet::weight(< T as Config >::WeightInfo::transfer())]
+		pub fn transfer(
+			origin: OriginFor<T>,
+			pool_id: PoolIdOf<T>,
+			tranche_id: TrancheIdOf<T>,
+			address: DomainAddress<Domain>,
+			amount: <T as pallet::Config>::Balance,
+		) -> DispatchResult {
+			let who = ensure_signed(origin.clone())?;
+
+			// Check that the destination is a member of this tranche token
+			ensure!(
+				T::Permission::has(
+					PermissionScope::Pool(pool_id),
+					address.into_account_truncating(),
+					Role::PoolRole(PoolRole::TrancheInvestor(tranche_id, Self::now()))
+				),
+				BadOrigin
+			);
+
+			ensure!(!amount.is_zero(), Error::<T>::InvalidTransferAmount);
+
+			// TODO: Transfer to the domain account for bookkeeping
+			// T::Tokens::transfer(
+			// 	T::CurrencyId::Tranche(pool_id, tranche_id),
+			// 	&who,
+			// 	&DomainLocator {
+			// 		domain: address.domain,
+			// 	}
+			// 	.into_account_truncating(),
+			// 	amount,
+			// 	false,
+			// )?;
+
+			Self::do_send_message(
+				who,
+				Message::Transfer {
+					pool_id,
+					tranche_id,
+					amount,
+					domain: address.clone().domain,
+					destination: address.clone().address,
+				},
+				address.domain,
+			)?;
+
+			Ok(())
+		}
+	}
+
+	impl<T: Config> Pallet<T> {
+		pub(crate) fn now() -> Moment {
+			T::Time::now().as_secs()
+		}
+
+		/// Send the `message` to the given domain.
+		pub fn do_send_message(
+			fee_payer: T::AccountId,
+			message: MessageOf<T>,
+			domain: Domain,
+		) -> DispatchResult {
+			let Router::Xcm(xcm_domain) =
+				<DomainRouter<T>>::get(domain.clone()).ok_or(Error::<T>::MissingRouter)?;
+
+			let contract_call = Self::encoded_contract_call(message.encode());
+			let ethereum_xcm_call =
+				Self::encoded_ethereum_xcm_call(xcm_domain.clone(), contract_call);
+
+			pallet_xcm_transactor::Pallet::<T>::transact_through_sovereign(
+				T::Origin::root(),
+				// The destination to which the message should be sent
+				Box::new(xcm_domain.location),
+				fee_payer,
+				// The currency in which we want to pay fees
+				CurrencyPayment {
+					currency: Currency::AsCurrencyId(xcm_domain.fee_currency),
```

Comment:
many currency... 

For my understanding, where does `Currency::AsCurrencyId` come from? And which currency enum are we referring to here? Ours?

Reply:
🤯 

Naturally...